.github/workflows/codeql-analysis.yml

```diff
@@ -42,6 +42,12 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
 
+      - name: setup-go
```

Comment:
We don't *have* to, but I if there is an official github action for the same task, then we should use it, to simplify our own code.

Reply:
sorry, I mean: why do we need to setup anything? codeql should be self-contained.

Reply:
If we don't we get @ianlewis had found that the Github action runner image still uses go1.20, which won't be able to recognize the new `toolchain` keyword in go.mod

>   2024/01/19 18:15:44 Error running go tooling: err: exit status 1: stderr: go: errors parsing go.mod:
  /home/runner/work/slsa-verifier/slsa-verifier/go.mod:5: unknown directive: toolchain
  ...
  2024/01/19 18:15:44 Extraction failed: exit status 1
  Error: We were unable to automatically build your code. Please replace the call to the autobuild action with your custom build steps. Encountered a fatal error while running "/opt/hostedtoolcache/CodeQL/2.16.0/x64/codeql/go/tools/autobuild.sh". Exit code was 1 and error was: 2024/01/19 18:15:44 Autobuilder was built with go1.21.5, environment has go1.20.12

Reply:
This is the "official" way to use the codeql Action. If you click thru GitHub settings, it will populate the exacts workflow. If it's failing, it's a bug on their side. Can we file a bug on their repo and ask?

Reply:
I should have included this as well:

Github's own advice for this known issue is to include a `setup-go` step. The `cache: false` isn't necessary, but it gets rid of the warnings.
 * https://github.com/github/codeql-action/issues/1842#issuecomment-1704398087

Reply:
LGTM, it's a know issue. Let's create an issue on our repo to keep track of this issue... just in case we can get rid of it in the future

Reply:
Alright #740